engine/cdn/storage/dao.go

```diff
@@ -267,24 +267,25 @@ func CountItemUnitByUnit(db gorp.SqlExecutor, unitID string) (int64, error) {
 	return db.SelectInt("SELECT COUNT(*) from storage_unit_item WHERE unit_id = $1", unitID)
 }
 
-func LoadAllItemIDUnknownByUnit(db gorp.SqlExecutor, unitID string, limit int) ([]string, error) {
+func LoadAllItemIDUnknownByUnitOrderByUnitID(db gorp.SqlExecutor, unitID string, orderUnitID string, limit int) ([]string, error) {
 	query := `
-		SELECT * 
+		SELECT id
 		FROM (
-			SELECT item.id 
+			SELECT item.id, storage_unit_item.unit_id
 			FROM item
 			JOIN storage_unit_item ON item.id = storage_unit_item.item_id
-			WHERE item.status = $3
-			EXCEPT 
-			SELECT item_id
-			FROM storage_unit_item  
-			WHERE unit_id = $1
+			WHERE 
+				item.status = $3 AND 
+				item.id NOT IN (SELECT item_id from storage_unit_item where unit_id = $1)
```

Comment:
NOT IN is not efficient and won't work if there is a lot of item in the storage_unit_item table.
That's by the Except was IMO a good solution.